src/irr/asset/CBufferLoaderBIN.cpp

```diff
@@ -0,0 +1,67 @@
+#include "CBufferLoaderBIN.h"
+
+namespace irr
+{
+	namespace asset
+	{
+		asset::SAssetBundle CBufferLoaderBIN::loadAsset(io::IReadFile* _file, const asset::IAssetLoader::SAssetLoadParams& _params, asset::IAssetLoader::IAssetLoaderOverride* _override, uint32_t _hierarchyLevel)
+		{
+			if (!_file)
+				return {};
+
+			SContext ctx;
+
+			ctx.files = _file;
+			ctx.files->grab();
+
+			uint16_t index{};
+
+			for (io::IReadFile* file = ctx.files; file; ++file)
+			{
+				ctx.sourceCodeBuffersCache.emplace_back(core::make_smart_refctd_ptr<char>(_IRR_NEW_ARRAY(char, file->getSize())));
+				file->read(ctx.sourceCodeBuffersCache[index++].get(), file->getSize());
+			}
+
+			return SAssetBundle(ctx.sourceCodeBuffersCache); // is it valid actually?
+		}
+
+		bool CBufferLoaderBIN::isALoadableFileFormat(io::IReadFile* _file) const
+		{
+			std::string filePath(_file->getFileName().c_str());
+			uint16_t beggingOfFileExtIndex{};
+
+			auto getIndexInArrayForFileExtension = [&](auto& index)
+			{
+				for (uint16_t i = filePath.size() - 1; i > 0u; --i)
+					if (filePath[i] == '.')
```

Comment:
we already have a function for getting the file extension, see how other loaders use it.